### PR TITLE
Improved: Display last-visited-time popup after login (OFBIZ-12148)

### PR DESCRIPTION
**Pull request by james-yong**
Improved: Display last-visited-time popup after login (OFBIZ-12148)

Popup last-visited time from database after user has logged in.

So users can know of any unauthorised access to their accounts.

### CODE REVIEW

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Locale;
+
+public class AfterLoginEvents {
+
+    private static final String MODULE = AfterLoginEvents.class.getName();
+    private static final String SCRIPT_SHOW_LAST_VISIT_DATE;
+
+    static {
+        SCRIPT_SHOW_LAST_VISIT_DATE = "<span id='showLastLogin'></span><script>"
+                + "importLibrary(%s, function () {\n"
```
Comment (mbrohl) [inline framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java:45]: This prevents the ability to translate the UI, we should at least use ui labels here.

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                        SimpleDateFormat formatter = new SimpleDateFormat("EE MMM d y H:m:s ZZZ");
+                        String dateString = formatter.format(fromDate);
+                        request.setAttribute("_UNSAFE_EVENT_MESSAGE_", String.format(SCRIPT_SHOW_LAST_VISIT_DATE, libJs, dateString));
+                    }
+                    count++;
+                }
+            }
+        } catch (GenericEntityException e) {
+            Debug.logError(e, MODULE);
+            request.setAttribute("_ERROR_MESSAGE_", e.getMessage());
```
Comment (mbrohl) [inline framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java]: This should not be an error message, it's just an information.

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                + "importLibrary(%s, function () {\n"
+                + "var dateFormat = Date.CultureInfo.formatPatterns.shortDate + ' ' + Date.CultureInfo.formatPatterns.longTime;\n"
+                + "var message = 'Your last visit was on '+new Date('%s').toString(dateFormat);\n"
+                + "$('#showLastLogin').replaceWith(message);\n"
+                + "});\n</script>";
+    }
+
+    public static String showLastVisit(HttpServletRequest request, HttpServletResponse response) {
+
+        // guard against re-popup while moving to other web application when tomcat SSO is enabled
```
Comment (mbrohl) [inline framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java:60]: We should also make the whole mechanism configurable (e.g. through a show.lastlogin.message = true|false configuration option).

**Review by mbrohl — CHANGES_REQUESTED** [review]
Thanks @james-yong ,

some thoughts inline.

The commit message should be amended to match our commit message template.

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java (hunk 1 above)
Comment (james-yong) [reply framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java:45]: Ok. Amended the script and added an entry in SecurityUiLabels.xml

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java (hunk 2 above)
Comment (james-yong) [reply framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java]: Removed the following:
request.setAttribute("_ERROR_MESSAGE_", e.getMessage());

File: framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java (hunk 3 above)
Comment (james-yong) [reply framework/webapp/src/main/java/org/apache/ofbiz/webapp/AfterLoginEvents.java:60]: Added "afterlogin.lastvisit.show" in security.properties

**Review by mbrohl — APPROVED** [review]

**Comment by james-yong** [comment]
Hi @mbrohl ,

Thanks for approval. 
Need your help to merge this pull request as I have no write access.

**Comment by JacquesLeRoux** [comment]
Wait James, you should have write access, you are a committer. Can't you merge here (in this case "Squash and merge") ?

**Comment by james-yong** [comment]
Hi Jacques,

I see the following for this pull request:
image

**Comment by JacquesLeRoux** [comment]
James, I think you need to follow https://servicecomb.apache.org/developers/setup-committer-rights/
Since above link is for Apache ServiceComb you may prefer to follow
https://www.allengeorge.com/2018/12/17/write-access-as-an-apache-committer/
And we could have a page like ServiceComb has in OFBiz site, ie at 
https://ofbiz.apache.org/developers/setup-committer-rights/

**Comment by james-yong** [comment]
Thanks Jacques. Followed the steps from the 1st link and now has the write access.

**Pull request merged**